### PR TITLE
Feature request 1

### PR DESCRIPTION
**Pull request by brucejh99**
- added modal component
- added duplicate checking with options to cancel or replace (should we also consider add?)

### CODE REVIEW

File: src/stores/BibliographyStore.js [hunk 1]
```diff
@@ -87,12 +88,17 @@ const BibliographyStoreModel = types
       }));
     },
     addCitation(newCitation) {
-      self.citations.push(newCitation);
-      self.latestId = newCitation.id;
-      localStorage.setItem(self.name, JSON.stringify({
-        style: self.style,
-        citations: self.citations,
-      }));
+      const isDuplicate = (self.citations.filter(citation => citation.url === newCitation.url).length > 0) ? true : false;
```
Comment (ddlan) [inline src/stores/BibliographyStore.js]: ternary is unnecessary here
also could just replace `isDuplicate` in the if condition with `self.citations.filter(citation => citation.url === newCitation.url).length > 0`

File: src/stores/BibliographyStore.js [hunk 2]
```diff
@@ -112,9 +118,17 @@ const BibliographyStoreModel = types
     setLatestId(id) {
       self.latestId = id;
     },
-    checkCitationUrlExists(url) {
-      return ((self.citations.filter(citation => citation.url === url)).length > 0);
-    },
+    resolveDuplicate(res, newCitation) {
+      if(res == 'update') {
```
Comment (ddlan) [inline src/stores/BibliographyStore.js]: use `===` here

File: src/stores/BibliographyStore.js (hunk 1 above)
Comment (brucejh99) [reply src/stores/BibliographyStore.js]: good point

File: src/stores/BibliographyStore.js (hunk 2 above)
Comment (brucejh99) [reply src/stores/BibliographyStore.js]: wow my js is rusty :(

**Pull request merged**